### PR TITLE
EWMA optimised for 2D datasets

### PR DESCRIPTION
**Pull request by rjenc29**
New Feature

- [x] Code conforms to style guide
- [ ] Doctests
- [x] Unit-tests
- [x] Documentation
- [x] New line item in [releases.md](https://github.com/fastats/fastats/blob/master/releases.md)
- [ ] Info added to `literature` directory

Bug Report

- [ ] Minimal repro in a test function

**Summary**
Additional EMWA function optimised for 2D data sets (updated the benchmarking notebook so you can see the relative gains).

Dense data sets only (no NaNs).

### CODE REVIEW

**Comment by dwillmer** [comment]
This is awesome, those timings are amazing :smile:

I was thinking of cutting a `2018.1` release tonight (for obvious reasons), did you want to get this merged ahead of that?

**Review by dwillmer — APPROVED** [review]
This is awesome, the speedup is massive and the test failures are spurious and from numba.

Happy to merge, we can either temporarily disable that test, or look at raising the numba version.

**Comment by rjenc29** [comment]
re: merge, it can wait - no big deal. :)

I didn't just blast the old implementation as I thought it would be more useful to be able to compare and contrast etc.  But that's why Codacy hates me (apparently dupe code etc etc)

**Comment by rjenc29** [comment]
We can fold it into a more general fn in the new year.

**Comment by dwillmer** [comment]
Sounds good - let me know when you're wanting a merge!

File: releases.md [hunk 1]
```diff
@@ -47,6 +47,7 @@ or eigenvectors.
 - drop_missing : helper function analogous to statsmodels missing='drop' mechanism which allows the user to evict 
 features and observations where one or more data points is not finite such that OLS may then be performed on dense / 
 complete data.
+- EWMA functions for 1D and 2D data sets (dense data / no NaNs)
```
Comment (dwillmer) [inline releases.md:50]: My bad - this will need moving to a 2019.2 section, but happy to do that in a later PR

Comment (rjenc29) [reply releases.md:50]: I think it's fine - the original EWMA function made the cut and it handles 1D and 2D dense data.  The addition in this PR is just a faster 2D-only version really.

Comment (dwillmer) [reply releases.md:50]: :thumbsup: nice one, will merge. we can add a `speedup for 2d ewma` enhancement message for 2019.2 later.

**Pull request merged**